frontend/src/Editor/Components/Table/Table.jsx

```diff
@@ -364,18 +364,39 @@ export function Table({
   }, [state.columnResizing.isResizingColumn]);
 
   const [paginationInternalPageIndex, setPaginationInternalPageIndex] = useState(pageIndex ?? 1);
-
+  const [rowDetails, setRowDetails] = useState();
   useEffect(() => {
     if (pageCount <= pageIndex) gotoPage(pageCount - 1);
   }, [pageCount]);
 
+  function usePrevious(value) {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  }
```

Comment:
This function seems to be called every time and the old value is set to ref every time it is called. Instead of this, can we use `ref` directly on the `onMouseOver`?

`onMouseOver={(e) => {
                      const hoveredRowDetails = { hoveredRowId: row.id, hoveredRow: row.original };
                      ref.current = rowDetails?.hoveredRowId;
                      setRowDetails(hoveredRowDetails);
                    }}`